cmd/metrics-scraper/app/scrape/scrape.go

```diff
@@ -0,0 +1,278 @@
+package scrape
+
+import(
+	"time"
+	"database/sql"
+	"log"
+	"sync"
+	"context"
+	 
+ 
+	"fmt"
+	"net/http"
+ 
+
+	"github.com/gin-gonic/gin"
+	 
+)
+
+var (
+    requests chan SaveRequest
+    db       *sql.DB
+    syncMap  sync.Map
+    // Add contexts and cancel functions for each app
+    appContexts     map[string]context.Context
+    appCancelFuncs  map[string]context.CancelFunc
+    contextMutex    sync.Mutex
+)
+
+
+func startAppMetricsFetcher(appName string) {
+	ticker := time.NewTicker(1 * time.Second)
```

Comment:
the ticker can be a option which our end-user can easily to config the polling interval, what do you think?

Reply:
Ya.. We can do that. we can modify the function to accept a duration parameter. This way, the user can specify the desired polling interval when calling the function.